### DISCUSSION TITLE
Where can I find a guide how to setup split-tuneling for Orbot and WireGuard to run simultaneously?

### DISCUSSION THREAD

**Discussion by onetimecontributor** [celzero/rethink-app#1789] [Q&A] [answered]
if I read it correctly here https://github.com/celzero/rethink-app?tab=readme-ov-file#vpn--proxifier 
` Split-tunneling further helps run multiple such tunnels at the same time and lets users route different apps over different tunnels`
and
` one could route Firefox over SOCKS5 connecting to Tor, Netflix over WireGuard connecting through any popular VPN provider, and Telegram or WhatsApp over censorship-resistant HTTP CONNECT endpoints at the same time.`

I do have existing config for orbot (manual config for both socks5 and http modes, tested OK) and wireguard (1 app selected for the wg, tested OK).
However I can't find the way to enable both orbot socks5 and wireguard to work simultaneously, rethink asks me to disable running proxy before I enable second one.

**Comment by ignoramous** [comment, marked as answer]
We had the feature but due to complications (mostly around *perceived* "leaks" some which really were users not being able to understand what exactly was being set up & what to expect), we removed this.

The UI needs to change significantly to not leave users confused if we are to bring this feature back again.

And since WireGuard is all we've been focusing on for over a year now, other proxies (HTTP, SOCKS5, Orbot) haven't got much attention (and are infact languishing at this point).

- #1105 
- #1724

**Reply by onetimecontributor** [reply]
tnx for clarification. it's a pity it turned out this way, please consider bringing it back! 
now it's not possible to balance anonymity and network performance that probably will push majority of "home" users to vpn-only (who will occasionally switch to Tor).

original use case "one could route Firefox over SOCKS5 connecting to Tor, Netflix over WireGuard" was just perfect representation of a real need:
1. Browser traffic is better "dissolved"/anonimised by Tor network, while
2.  heavy media streams, (where end user is "anyway" at least somehow "known" to the streaming platform), should go to VPN to address protection from network provider and also to reduce some fingerprint for streaming platform, like exact location of your devices (as city is known, etc)

I've seen #1724, but it is a totally different use case for me - on a much deeper level of configuration (more for convenience) comparing to a higher-level need to separate and balance level of anonymity vs performance.

**Reply by onetimecontributor** [reply]
is there any way to bring it back disabled by default and only under some "advanced" section with disclaimer? 
I know the fact I guess my ticket is a kind of rare one, that may speak for itself, but still...

**Reply by ignoramous** [reply]
Not possible, as we've removed the code (and as I mentioned above, the WireGuard related code has deviated so much that the other proxies are languishing behind). It isn't straight forward to bring all proxy types to parity with WireGuard, but intend to make it happen after the next version `v055o` is out.

**Reply by onetimecontributor** [reply]
> Not possible

@ignoramous, Do you by chance know if I can workaround such limitation using Orbot 's "power user mode" (allows you to start Orbot without VPN setting and shows open SOCKS and HTTP ports)? I've tried to configure Orbot this way and set Cromite to use it as local Proxy (HTTP or PROXY 127.0.0.1:8118) but due to a bug in Cromite https://github.com/uazo/cromite/issues/1627 I coudn't get it working.
So atm I'm wondering if at least there is a chance to use RethinkDNS in WireGuard mode and Orbot on top (not managed by RethinkDNS app) this way and thus seek your advise from your best knowledge. Tx

**Reply by ignoramous** [reply]
Correct. That setup should work as you expect it to.

**Reply by onetimecontributor** [reply]
> Correct. That setup should work as you expect it to.

great, tnx, I'll find FOSS browser that supports proxy config an confirm. Though it's not a complete workaround, but at least I will be able to use Tor over VPN for apps that supports it.

**Reply by onetimecontributor** [reply]
> Correct. That setup should work as you expect it to.

yep, confirmed via Mull+FoxyProxy over Orbot (non-VPN SOCKS5 mode)

**Reply by onetimecontributor** [reply]
great, so the workaround works also over protonVPN
so once I get help with rdns #1793 (how to properly configure rethinkDNS to use NextDNS while using WireGuard), ~~and cromite fixes https://github.com/uazo/cromite/issues/1627~~ , I'm covered. Thank you for your time!

**Reply by onetimecontributor** [reply]
> great, so the workaround works also over protonVPN so once I get help with rdns #1793 ~and cromite fixes [uazo/cromite#1627](https://github.com/uazo/cromite/issues/1627)~ , I'm covered. Thank you for your time!

EDIT: it appeared there is no bug in Cromite re proxy that affects me, it was something I did wrong or coincidence.
So once I get help with #1793 (how to properly configure rethinkDNS to use NextDNS while using WireGuard), I'm covered for at least browsers part via Tor
